### PR TITLE
Jim/76934/tabs component ts migration

### PR DESCRIPTION
**Pull request by jim-deriv**
## Changes:

Please include a summary of the change and which issue is fixed below:
-   ...

## When you need to add unit test

-   [ ] If this change disrupt current flow
-   [ ] If this change is adding new flow

## When you need to add integration test

-   [ ] If components from external libraries are being used to define the flow, e.g. @deriv/components
-   [ ] If it relies on a very specific set of props with no default behavior for the current component.

## Test coverage checklist (for reviewer)

-   [ ] Ensure utility / function has a test case 
-   [ ] Ensure all the tests are passing

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [X] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: packages/components/src/components/tabs/tabs.tsx [hunk 1]
```diff
@@ -1,39 +1,63 @@
...
 const Tabs = ({
     active_icon_color,
-    active_index,
+    active_index = 0,
     background_color,
     bottom,
     center,
     children,
-    className,
+    className = '',
```
Comment (yauheni-deriv) [inline packages/components/src/components/tabs/tabs.tsx]: why do we need default value here? we can just skip it, and for Tab component mark also this className property like optional, as we do it everywhere in the code. How do you think? :)

File: packages/components/src/components/tabs/tabs.tsx [hunk 2]
```diff
@@ -52,10 +76,11 @@ const Tabs = ({
         }
     }, []);
 
-    let initial_index_to_show, tab_width;
+    let initial_index_to_show = 0;
```
Comment (yauheni-deriv) [inline packages/components/src/components/tabs/tabs.tsx:79]: why did you change it here? as I see, you set for active_index default value 0, so in the 83 line initial_index_to_show anyway will be at list 0. no needs to set it here, I guess..

File: packages/components/src/components/tabs/tabs.tsx [hunk 3]
```diff
@@ -52,10 +76,11 @@ const Tabs = ({
...
+    let tab_width;
```
Comment (farzin-deriv) [inline packages/components/src/components/tabs/tabs.tsx]:
```suggestion
    let tab_width: string;
```
@jim-deriv Not sure tho 🙇🏻

**Review by farzin-deriv — APPROVED** [review]

File: packages/components/src/components/tabs/tabs.tsx (hunk 2 above)
Comment (jim-deriv) [reply packages/components/src/components/tabs/tabs.tsx:79]: @yauheni-kryzhyk-deriv, as seen in the screenshot attached here, there is a scoping issue I think (`Variable 'initial_index_to_show' is used before being assigned.ts(2454)`), the assignment inside the `useConstructor` hook is not recognised, so I have to initialise it regardless of what  I do inside the hook

before_assignment

**Review by farzin-deriv — APPROVED** [review]

File: packages/components/src/components/themed-scrollbars/themed-scrollbars.tsx [hunk 4]
```diff
@@ -28,13 +28,13 @@ const ThemedScrollbars = ({
     is_only_horizontal_overlay,
     is_scrollbar_hidden,
     onScroll,
-    refSetter,
+    refSetter = null,
```
Comment (niloofar-deriv) [inline packages/components/src/components/themed-scrollbars/themed-scrollbars.tsx:31]: Do we need to pass `null` to this?

Comment (yauheni-deriv) [reply packages/components/src/components/themed-scrollbars/themed-scrollbars.tsx:31]: @niloo-fs I think yes, we have to pass into ref first value in order to not to pass undefined there, as this prop is optional

File: packages/components/src/components/tabs/tabs.tsx (hunk 2 above)
Comment (yauheni-deriv) [reply packages/components/src/components/tabs/tabs.tsx:79]: Ok, understand it. So maybe it would be better decision to pass to useState initial value 0, and and left that variable unassigned as before?. Just  for avoiding some confusions while reading the code and seeing that we are assigning  0 to this variable two times :)
how do you think?

Comment (jim-deriv) [reply packages/components/src/components/tabs/tabs.tsx:79]: Alright. I'll explore that and see

**Review by ashraf-deriv — APPROVED** [review]

File: packages/components/src/components/themed-scrollbars/themed-scrollbars.tsx (hunk 4 above)
Comment (niloofar-deriv) [reply packages/components/src/components/themed-scrollbars/themed-scrollbars.tsx:31]: Got it thank u @yauheni-kryzhyk-deriv

**Review by niloofar-deriv — APPROVED** [review]

**Review by shayan-deriv — APPROVED** [review]
